src/log/log_slot.c

```diff
@@ -412,6 +412,12 @@ __wt_log_slot_switch(
         WT_RET(WT_SESSION_CHECK_PANIC(session));
         if (F_ISSET(S2C(session), WT_CONN_CLOSING))
             break;
+        /*
+         * If we get an unexpected error, we need to panic. If we cannot switch the slot because of
+         * a real error, such as running out of space, there's nothing we can do.
+         */
+        if (ret != 0 && ret != EBUSY)
+            return (__wt_panic(session, ret, "log slot switch fatal error"));
```

Comment:
I was wondering if we might want this new check above the check of `WT_CONN_CLOSING`?  Thinking about the situation that we are in the midst of closing the connection and happen to get a fatal error in logging - we want to make sure the error isn't lost.  But I think it amounts to the same thing, the `break` means we'll return the error (not a panic), and callers from higher up will report a failed log write.

Reply:
I think that is a good idea. I kept it after checking if the PANIC flag is already set, but checking before close is good.

Reply:
I guess I have a similar thought about moving this earlier, but for different reasons from Don.

If you put this check after the `ret == EBUSY` check then you can simplify the condition by making it an `else if (ret != 0)`, and you don't need to explicitly call return as the following `WT_RET(WT_SESSION_CHECK_PANIC(session))` will do that for you.  

I think this is mostly an aesthetic change, so I'll leave it to your judgement whether it is desirable.